app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/Model/DummyDataSetInterface.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.HumanLearning2021.HumanLearningApp.Model
+
+import java.lang.IllegalArgumentException
+
+/**
+ * a class representing the dummy interface to a data set
+ */
+class DummyDataSetInterface: DataSetInterface {
+    val currentDataSet = DummyDataSet()
+
+    override suspend fun getPicture(categoryString: String): CategorizedPicture {
+        return currentDataSet.getPicture(getCategory(categoryString))
+
+    }
+
+
+    private fun getCategory(name: String):Category {
+        for(category in currentDataSet.categories) {
+            if(category.name.equals(name, ignoreCase = true)) return category
```

Comment:
Can't you reuse the equals method you defined in Category?
```suggestion
            if(category.equals(Category(name))) return category
```

Reply:
Strictly speaking yes, however if Category were to receive more fields later this would break, so it's not very future-proof.